__tests__/app.test.js

```diff
@@ -330,3 +330,127 @@ describe("GET /api/articles/:article_id/comments", () => {
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+    test("invalid value returns 400 bad request", () => {
+      return request(app)
+        .get("/api/articles?sort_by=nonsense")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad request");
+        });
+    });
+  });
+  describe("order", () => {
+    test("can order by ASC", () => {
+      return request(app)
+        .get("/api/articles?order=ASC")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy("created_at", {
+            ascending: true,
+          });
+        });
+    });
+    test("can order by DESC", () => {
+      return request(app)
+        .get("/api/articles?order=DESC")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+    test("if no input, defaults to desc", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+    test("invalid input value returns 400 bad request", () => {
+      return request(app)
+        .get("/api/articles?order=nonsense")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad request");
+        });
+    });
+  });
+  describe("topic", () => {
+    test("should filter by input topic cats", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles[0].topic).toBe("cats");
+        });
+    });
+    test("filters for mitch", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBeGreaterThan(1);
+          body.articles.forEach((article) => {
+            expect(article.topic).toBe("mitch");
+          });
+        });
+    });
+    test("if no input, no articles are filtered out", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBeGreaterThan(1);
+          let mitchCount = 0;
+          let catsCount = 0;
+          body.articles.forEach((article) => {
+            if (article.topic === "cats") {
+              catsCount++;
+            } else if (article.topic === "mitch") {
+              mitchCount++;
+            }
+          });
+          expect(mitchCount).not.toBe(0);
+          expect(catsCount).not.toBe(0);
+          //is there a better way to do this ^^^ (checking both mitch and cats exist)
```

Comment:
I think you'd get away with just checking that article.length is the number of all articles present in the test db, no need to do all the other checks